{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Playground"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Resources:\n",
    "1. http://biopython.org/DIST/docs/tutorial/Tutorial.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "# Imports.\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from utility_functions import desc_df\n",
    "from utility_functions import print_color\n",
    "\n",
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>code3letter</th>\n",
       "      <th>name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>code1letter</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A</th>\n",
       "      <td>Ala</td>\n",
       "      <td>Alanine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C</th>\n",
       "      <td>Cys</td>\n",
       "      <td>Cysteine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>D</th>\n",
       "      <td>Asp</td>\n",
       "      <td>Aspartic Acid</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E</th>\n",
       "      <td>Glu</td>\n",
       "      <td>Glutamic Acid</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>F</th>\n",
       "      <td>Phe</td>\n",
       "      <td>Phenylalanine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G</th>\n",
       "      <td>Gly</td>\n",
       "      <td>Glycine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H</th>\n",
       "      <td>His</td>\n",
       "      <td>Histidine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>I</th>\n",
       "      <td>Ile</td>\n",
       "      <td>Isoleucine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>K</th>\n",
       "      <td>Lys</td>\n",
       "      <td>Lysine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L</th>\n",
       "      <td>Leu</td>\n",
       "      <td>Leucine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M</th>\n",
       "      <td>Met</td>\n",
       "      <td>Methionine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N</th>\n",
       "      <td>Asn</td>\n",
       "      <td>Asparagine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P</th>\n",
       "      <td>Pro</td>\n",
       "      <td>Proline</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q</th>\n",
       "      <td>Gln</td>\n",
       "      <td>Glutamine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R</th>\n",
       "      <td>Arg</td>\n",
       "      <td>Arginine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S</th>\n",
       "      <td>Ser</td>\n",
       "      <td>Serine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T</th>\n",
       "      <td>Thr</td>\n",
       "      <td>Threonine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V</th>\n",
       "      <td>Val</td>\n",
       "      <td>Valine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>W</th>\n",
       "      <td>Trp</td>\n",
       "      <td>Tryptophan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Y</th>\n",
       "      <td>Tyr</td>\n",
       "      <td>Tyrosine</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            code3letter           name\n",
       "code1letter                           \n",
       "A                   Ala        Alanine\n",
       "C                   Cys       Cysteine\n",
       "D                   Asp  Aspartic Acid\n",
       "E                   Glu  Glutamic Acid\n",
       "F                   Phe  Phenylalanine\n",
       "G                   Gly        Glycine\n",
       "H                   His      Histidine\n",
       "I                   Ile     Isoleucine\n",
       "K                   Lys         Lysine\n",
       "L                   Leu        Leucine\n",
       "M                   Met     Methionine\n",
       "N                   Asn     Asparagine\n",
       "P                   Pro        Proline\n",
       "Q                   Gln      Glutamine\n",
       "R                   Arg       Arginine\n",
       "S                   Ser         Serine\n",
       "T                   Thr      Threonine\n",
       "V                   Val         Valine\n",
       "W                   Trp     Tryptophan\n",
       "Y                   Tyr       Tyrosine"
      ]
     },
     "execution_count": 253,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_iupac_aa_code = pd.read_csv('../Data/__iupac_amino_acid_code.csv')\n",
    "df_iupac_aa_code.set_index('code1letter', inplace=True, drop=True)\n",
    "df_iupac_aa_code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "metadata": {},
   "outputs": [],
   "source": [
    "proteins = {\"id\":[], \"name\":[], \"source\":[], \"plastic\":[], \"n_plastics\":[], \"aa_seq\":[]}\n",
    "for p_data in SeqIO.parse(\"../Data/__plasticdb_protein_sequences.fasta\", \"fasta\"):\n",
    "    p_id, name, source, plastic = p_data.id.lower().replace('-','_').split(\"||\")\n",
    "    aa_seq = p_data.seq\n",
    "    proteins['id'].append(int(p_id))\n",
    "    proteins['name'].append(name)\n",
    "    proteins['source'].append(source)\n",
    "    proteins['plastic'].append(plastic)\n",
    "    proteins['n_plastics'].append(len(plastic.split(\"_\")))\n",
    "    proteins['aa_seq'].append(aa_seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>source</th>\n",
       "      <th>plastic</th>\n",
       "      <th>n_plastics</th>\n",
       "      <th>aa_seq</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>phb_depolymerase</td>\n",
       "      <td>ralstonia_pickettii</td>\n",
       "      <td>phb_pha</td>\n",
       "      <td>2</td>\n",
       "      <td>(M, K, H, P, Y, G, Y, R, W, H, W, L, Y, A, L, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>pha_depolymerase</td>\n",
       "      <td>pseudomonas_fluorescens</td>\n",
       "      <td>pho_pha</td>\n",
       "      <td>2</td>\n",
       "      <td>(M, P, L, R, T, L, L, C, G, L, L, L, A, V, C, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>phb_depolymerase_a</td>\n",
       "      <td>paucimonas_lemoignei</td>\n",
       "      <td>phb_pha</td>\n",
       "      <td>2</td>\n",
       "      <td>(M, R, N, T, L, K, A, A, F, K, L, G, V, I, S, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>phb_depolymerase</td>\n",
       "      <td>paucimonas_lemoignei</td>\n",
       "      <td>phb_pha</td>\n",
       "      <td>2</td>\n",
       "      <td>(M, L, A, K, Q, I, K, K, A, N, S, R, S, T, L, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>pha_depolymerase_b</td>\n",
       "      <td>paucimonas_lemoignei</td>\n",
       "      <td>phb_pha</td>\n",
       "      <td>2</td>\n",
       "      <td>(M, M, S, S, Q, T, T, Q, S, S, K, F, S, L, F, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204</th>\n",
       "      <td>petase</td>\n",
       "      <td>thermobifida_fusca</td>\n",
       "      <td>pet</td>\n",
       "      <td>1</td>\n",
       "      <td>(M, A, A, N, P, Y, E, R, G, P, N, P, T, D, A, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>petase</td>\n",
       "      <td>thermobifida_alba</td>\n",
       "      <td>pet</td>\n",
       "      <td>1</td>\n",
       "      <td>(M, A, N, P, Y, E, R, G, P, N, P, T, E, S, M, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>petase</td>\n",
       "      <td>thermobifida_cellulosilytica</td>\n",
       "      <td>pet</td>\n",
       "      <td>1</td>\n",
       "      <td>(M, A, N, P, Y, E, R, G, P, D, P, T, Q, A, S, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>petase</td>\n",
       "      <td>thermobifida_halotolerans</td>\n",
       "      <td>pet</td>\n",
       "      <td>1</td>\n",
       "      <td>(M, A, N, P, Y, E, R, G, P, N, P, T, N, S, S, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>petase</td>\n",
       "      <td>thermobifida_cellulosilytica</td>\n",
       "      <td>pet</td>\n",
       "      <td>1</td>\n",
       "      <td>(M, A, N, P, Y, E, R, G, P, N, P, T, D, A, L, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>208 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   name                        source  plastic  n_plastics  \\\n",
       "id                                                                           \n",
       "1      phb_depolymerase           ralstonia_pickettii  phb_pha           2   \n",
       "2      pha_depolymerase       pseudomonas_fluorescens  pho_pha           2   \n",
       "3    phb_depolymerase_a          paucimonas_lemoignei  phb_pha           2   \n",
       "4      phb_depolymerase          paucimonas_lemoignei  phb_pha           2   \n",
       "5    pha_depolymerase_b          paucimonas_lemoignei  phb_pha           2   \n",
       "..                  ...                           ...      ...         ...   \n",
       "204              petase            thermobifida_fusca      pet           1   \n",
       "205              petase             thermobifida_alba      pet           1   \n",
       "206              petase  thermobifida_cellulosilytica      pet           1   \n",
       "207              petase     thermobifida_halotolerans      pet           1   \n",
       "208              petase  thermobifida_cellulosilytica      pet           1   \n",
       "\n",
       "                                                aa_seq  \n",
       "id                                                      \n",
       "1    (M, K, H, P, Y, G, Y, R, W, H, W, L, Y, A, L, ...  \n",
       "2    (M, P, L, R, T, L, L, C, G, L, L, L, A, V, C, ...  \n",
       "3    (M, R, N, T, L, K, A, A, F, K, L, G, V, I, S, ...  \n",
       "4    (M, L, A, K, Q, I, K, K, A, N, S, R, S, T, L, ...  \n",
       "5    (M, M, S, S, Q, T, T, Q, S, S, K, F, S, L, F, ...  \n",
       "..                                                 ...  \n",
       "204  (M, A, A, N, P, Y, E, R, G, P, N, P, T, D, A, ...  \n",
       "205  (M, A, N, P, Y, E, R, G, P, N, P, T, E, S, M, ...  \n",
       "206  (M, A, N, P, Y, E, R, G, P, D, P, T, Q, A, S, ...  \n",
       "207  (M, A, N, P, Y, E, R, G, P, N, P, T, N, S, S, ...  \n",
       "208  (M, A, N, P, Y, E, R, G, P, N, P, T, D, A, L, ...  \n",
       "\n",
       "[208 rows x 5 columns]"
      ]
     },
     "execution_count": 255,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_proteins = pd.DataFrame(proteins)\n",
    "df_proteins.set_index('id', drop=True, inplace=True)\n",
    "df_proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "197\n",
      "aa_seq\n",
      "(M, A, V, M, T, P, R, R, E, R, S, S, L, L, S, R, A, L, Q, V, T, A, A, A, A, T, A, L, V, T, A, V, S, L, A, A, P, A, H, A, A, N, P, Y, E, R, G, P, N, P, T, D, A, L, L, E, A, S, S, G, P, F, S, V, S, E, E, N, V, S, R, L, S, A, S, G, F, G, G, G, T, I, Y, Y, P, R, E, N, N, T, Y, G, A, V, A, I, S, P, G, Y, ...)    3\n",
      "(M, R, S, I, R, L, K, R, L, I, A, A, V, A, L, G, G, A, A, A, A, T, Q, A, A, S, P, L, P, R, L, N, V, D, K, T, Q, I, S, V, S, G, L, S, A, G, G, F, M, A, N, Q, L, G, Y, A, Y, S, G, T, F, M, G, V, G, I, F, A, G, G, P, Y, M, C, A, G, H, S, N, Y, T, S, C, M, Y, N, A, T, I, T, S, S, M, R, S, A, M, Q, A, S, ...)    3\n",
      "(M, G, V, F, D, Y, K, N, F, T, A, S, D, S, K, A, L, F, S, D, A, L, A, I, T, L, Y, S, Y, H, N, I, D, N, G, F, A, E, G, Y, Q, H, N, G, F, G, L, G, L, P, A, T, L, V, T, A, L, I, G, S, G, N, S, Q, G, V, I, P, G, I, P, W, N, P, D, S, E, K, A, A, L, D, A, L, H, Q, A, G, W, S, T, I, S, A, Q, Q, L, G, Y, D, ...)    2\n",
      "(M, N, F, P, R, A, S, R, L, M, Q, A, A, V, L, G, G, L, M, A, V, S, A, A, A, T, A, Q, T, N, P, Y, A, R, G, P, N, P, T, A, A, S, L, E, A, S, A, G, P, F, T, V, R, S, F, T, V, S, R, P, S, G, Y, G, A, G, T, V, Y, Y, P, T, N, A, G, G, T, V, G, A, I, A, I, V, P, G, Y, T, A, R, Q, S, S, I, K, W, W, G, P, R, ...)    2\n",
      "(M, R, G, V, W, R, Y, M, P, V, Y, Y, Y, K, C, M, L, I, P, L, L, T, F, T, K, I, S, I, I, S, K, P, Q, E, S, R, I, S, L, I, I, C, R, T, M, P, T, V, A, T, F, L, G, L, A, I, S, G, A, A, M, A, S, G, A, A, L, G, A, Y, N, V, D, P, N, S, I, S, V, S, G, L, S, S, G, G, F, M, S, A, Q, L, G, V, A, Y, S, D, T, F, ...)    2\n",
      "                                                                                                                                                                                                                                                                                                                    ..\n",
      "(M, K, K, V, I, V, F, L, V, F, M, L, F, C, T, A, T, Y, A, Q, K, V, S, K, V, V, F, F, G, D, S, L, T, D, N, G, N, L, F, R, L, V, E, L, I, P, K, T, P, P, Y, Y, Q, G, R, F, S, N, G, P, T, W, A, E, N, L, G, E, Y, F, V, K, T, N, G, A, K, Y, E, I, Y, A, Y, A, G, A, T, A, A, A, H, N, I, F, E, T, L, A, P, T, ...)    1\n",
      "(M, N, E, A, M, I, R, G, V, L, E, E, A, V, A, S, G, G, I, A, G, A, V, V, T, V, T, D, A, N, A, T, L, F, Q, L, A, V, G, R, Q, S, V, A, A, P, D, P, M, R, A, D, S, V, F, W, I, A, S, M, T, K, A, I, T, T, V, A, A, L, Q, L, V, E, Q, G, K, L, T, L, D, E, P, I, G, D, V, L, P, E, L, A, E, P, V, V, L, K, G, F, ...)    1\n",
      "(M, R, T, V, D, G, R, I, G, M, V, E, R, R, T, R, L, V, S, L, L, A, A, A, A, L, A, V, S, L, S, G, C, S, A, T, T, V, L, N, T, L, E, P, R, W, D, V, A, A, T, R, D, V, A, Y, A, P, G, P, R, H, D, V, D, V, Y, A, P, K, K, I, G, P, H, T, P, V, V, V, F, I, Y, G, G, G, W, D, S, G, E, K, S, K, Y, A, F, V, G, S, ...)    1\n",
      "(M, T, T, R, P, E, S, L, G, L, S, A, E, R, L, K, R, L, D, A, H, F, D, A, K, Y, V, Q, T, G, K, L, P, F, A, M, T, L, V, S, R, K, G, E, V, A, H, H, G, F, S, G, L, M, D, E, A, R, G, K, K, V, E, A, D, S, L, F, R, I, Y, S, M, T, K, P, L, T, S, V, L, F, M, M, L, V, E, E, G, K, V, A, L, D, D, P, V, H, S, L, ...)    1\n",
      "(M, A, N, P, Y, E, R, G, P, N, P, T, D, A, L, L, E, A, R, S, G, P, F, S, V, S, E, E, N, V, S, R, F, G, A, D, G, F, G, G, G, T, I, Y, Y, P, R, E, N, N, T, Y, G, A, V, A, I, S, P, G, Y, T, G, T, Q, A, S, V, A, W, L, G, E, R, I, A, S, H, G, F, V, V, I, T, I, D, T, N, T, T, L, D, Q, P, D, S, R, A, R, Q, ...)    1\n",
      "Name: count, Length: 197, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "vc = df_proteins['aa_seq'].value_counts()\n",
    "print(len(vc))\n",
    "print(vc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 257,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(108, 914)"
      ]
     },
     "execution_count": 257,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "min(df_proteins['aa_seq'].apply(len)), max(df_proteins['aa_seq'].apply(len))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No. of rows = 1701\n",
      "No. of columns = 22\n",
      "\u001b[33m\n",
      "Feature Overview\u001b[0m\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1701 entries, 0 to 1700\n",
      "Data columns (total 22 columns):\n",
      " #   Column                                Non-Null Count  Dtype  \n",
      "---  ------                                --------------  -----  \n",
      " 0   Microorganism                         1701 non-null   object \n",
      " 1   Tax ID                                1701 non-null   int64  \n",
      " 2   Plastic                               1701 non-null   object \n",
      " 3   Ref                                   1701 non-null   object \n",
      " 4   Enzyme                                1654 non-null   object \n",
      " 5   Enzyme ID                             384 non-null    float64\n",
      " 6   Database Enzyne Name                  384 non-null    object \n",
      " 7   Gene                                  1691 non-null   object \n",
      " 8   GenbankID                             320 non-null    object \n",
      " 9   Sequence                              384 non-null    object \n",
      " 10  Year                                  1700 non-null   float64\n",
      " 11  Evidence                              1656 non-null   object \n",
      " 12  Plastic used                          1574 non-null   object \n",
      " 13  Manufacturer                          1232 non-null   object \n",
      " 14  Analitical grade                      949 non-null    object \n",
      " 15  Thermophilic conditions               1647 non-null   object \n",
      " 16  Isolation sample type                 1211 non-null   object \n",
      " 17  Isolation environment                 1265 non-null   object \n",
      " 18  Isolation location                    1048 non-null   object \n",
      " 19  Degradation extrapolated from enzyme  1696 non-null   object \n",
      " 20  Paper id                              24 non-null     object \n",
      " 21  lineage                               1701 non-null   object \n",
      "dtypes: float64(2), int64(1), object(19)\n",
      "memory usage: 292.5+ KB\n",
      "None\n",
      "\u001b[33m\n",
      "Value Counts\u001b[0m\n",
      "\n",
      "Microorganism\n",
      "Uncultured bacterium           57\n",
      "Streptomyces sp.               42\n",
      "Aspergillus niger              40\n",
      "Pseudomonas sp.                38\n",
      "Bacillus sp.                   27\n",
      "                               ..\n",
      "Saccharothrix cryophilis        1\n",
      "Saccharothrix coeruleofusca     1\n",
      "Saccharothrix australiensis     1\n",
      "Rhodotorula sphaerocarpa        1\n",
      "Saccharopolyspora flava         1\n",
      "Name: count, Length: 614, dtype: int64\n",
      "No. of distinct values = 614\n",
      "No. of missing values = 0\n",
      "\n",
      "Tax ID\n",
      "77133    57\n",
      "1931     43\n",
      "5061     40\n",
      "306      38\n",
      "1409     28\n",
      "         ..\n",
      "5078      1\n",
      "69773     1\n",
      "63821     1\n",
      "63820     1\n",
      "95161     1\n",
      "Name: count, Length: 586, dtype: int64\n",
      "No. of distinct values = 586\n",
      "No. of missing values = 0\n",
      "\n",
      "Plastic\n",
      "PU              309\n",
      "PHB             212\n",
      "PCL             162\n",
      "PLA             155\n",
      "LDPE            145\n",
      "               ... \n",
      "Treated-HDPE      1\n",
      "PLA Blend         1\n",
      "PCL Blend         1\n",
      "PBS Blend         1\n",
      "PU Blend          1\n",
      "Name: count, Length: 64, dtype: int64\n",
      "No. of distinct values = 64\n",
      "No. of missing values = 0\n",
      "\n",
      "Ref\n",
      "Navarro, D., Chaduli, D., Taussac, S., Lesage-Meessen, L., Grisel, S., Haon, M., ... & Favel, A. (2021). Large-scale phenotyping of 1,000 fungal strains for the degradation of non-natural, industrial compounds.                                                                                                                 220\n",
      "Jeszeová, L., Puškárová, A., Bučková, M., Kraková, L., Grivalský, T., Danko, M., ... & Pangallo, D. (2018). Microbial communities responsible for the degradation of poly (lactic acid)/poly (3-hydroxybutyrate) blend mulches in soil burial respirometric tests. World Journal of Microbiology and Biotechnology, 34(7), 101.     52\n",
      "Suyama, T., Tokiwa, Y., Ouichanpagdee, P., Kanagawa, T., & Kamagata, Y. (1998). Phylogenetic affiliation of soil bacteria that degrade aliphatic polyesters available commercially as biodegradable plastics. Appl. Environ. Microbiol., 64(12), 5008-5011.                                                                         38\n",
      "Pranamuda, H., & Tokiwa, Y. (1999). Degradation of poly (L-lactide) by strains belonging to genus Amycolatopsis. Biotechnology Letters, 21(10), 901-905.                                                                                                                                                                            37\n",
      "Gangoiti, J., Santos, M., Prieto, M. A., de la Mata, I., Serra, J. L., & Llama, M. J. (2012). Characterization of a novel subgroup of extracellular medium-chain-length polyhydroxyalkanoate depolymerases from actinobacteria. Appl. Environ. Microbiol., 78(20), 7229-7237.                                                       33\n",
      "                                                                                                                                                                                                                                                                                                                                  ... \n",
      "Watanabe, T., Ohtake, Y., Asabe, H., Murakami, N., & Furukawa, M. (2009). Biodegradability and degrading microbes of low‐density polyethylene. Journal of applied polymer science, 111(1), 551-559.                                                                                                                                  1\n",
      "Jarerat, A., & Tokiwa, Y. (2003). Poly (L-lactide) degradation by Saccharothrix waywayandensis. Biotechnology letters, 25(5), 401-404.                                                                                                                                                                                               1\n",
      "Volke‐Sepúlveda, T., Saucedo‐Castañeda, G., Gutiérrez‐Rojas, M., Manzur, A., & Favela‐Torres, E. (2002). Thermally treated low density polyethylene biodegradation by Penicillium pinophilum and Aspergillus niger. Journal of Applied Polymer Science, 83(2), 305-314.                                                              1\n",
      "Azeko, S. T., Etuk-Udo, G. A., Odusanya, O. S., Malatesta, K., Anuku, N., & Soboyejo, W. O. (2015). Biodegradation of linear low density polyethylene by Serratia marcescens subsp. marcescens and its cell free extracts. Waste and biomass valorization, 6(6), 1047-1057.                                                          1\n",
      "Skariyachan, S., Megha, M., Kini, M. N., Mukund, K. M., Rizvi, A., & Vasist, K. (2015). Selection and screening of microbial consortia for efficient and ecofriendly degradation of plastic garbage collected from urban and rural areas of Bangalore, India. Environmental monitoring and assessment, 187(1), 4174.                 1\n",
      "Name: count, Length: 473, dtype: int64\n",
      "No. of distinct values = 473\n",
      "No. of missing values = 0\n",
      "\n",
      "Enzyme\n",
      "No                        1200\n",
      "PHB depolymerase            82\n",
      "Cutinase                    68\n",
      "Esterase                    45\n",
      "Hydrolase                   35\n",
      "                          ... \n",
      "Alkane-1 monooxygenase       1\n",
      "Polyurethanase A             1\n",
      "Polyurethanase B             1\n",
      "MHETase                      1\n",
      "Polyamidase                  1\n",
      "Name: count, Length: 61, dtype: int64\n",
      "No. of distinct values = 61\n",
      "No. of missing values = 47\n",
      "\n",
      "Enzyme ID\n",
      "124.0    9\n",
      "76.0     8\n",
      "74.0     8\n",
      "47.0     7\n",
      "49.0     7\n",
      "        ..\n",
      "106.0    1\n",
      "107.0    1\n",
      "108.0    1\n",
      "109.0    1\n",
      "208.0    1\n",
      "Name: count, Length: 208, dtype: int64\n",
      "No. of distinct values = 208\n",
      "No. of missing values = 1317\n",
      "\n",
      "Database Enzyne Name\n",
      "00124 | Polyester hydrolase | Moraxella sp. | PET PCL PU PHB PBS PLA PHA            9\n",
      "00076 | Cutinase | Saccharomonospora viridis | PBAT PBS PBSA PCL PET PHB PLA PHA    8\n",
      "00074 | Cutinase | Saccharomonospora viridis | PBAT PBS PBSA PCL PET PHB PLA PHA    8\n",
      "00047 | PLA depolymerase | Uncultured bacterium | PLA PBS PBSA PCL PES PHB PHA      7\n",
      "00049 | PLA depolymerase | Uncultured bacterium | PBS PBSA PCL PES PHB PLA PHA      7\n",
      "                                                                                   ..\n",
      "00106 | Cutinase | Thermobifida fusca | PET                                         1\n",
      "00107 | Cutinase | Thermobifida alba | PET                                          1\n",
      "00108 | Serine hydrolase | Thermobifida halotolerans | PET                          1\n",
      "00109 | Triacylglycerol lipase | Polyangium brachysporum | PET                      1\n",
      "00208 | PETase | Thermobifida cellulosilytica | PET                                 1\n",
      "Name: count, Length: 208, dtype: int64\n",
      "No. of distinct values = 208\n",
      "No. of missing values = 1317\n",
      "\n",
      "Gene\n",
      "No     1307\n",
      "Yes     384\n",
      "Name: count, dtype: int64\n",
      "No. of distinct values = 2\n",
      "No. of missing values = 10\n",
      "\n",
      "GenbankID\n",
      "P19833        9\n",
      "BAA19791.1    8\n",
      "BAO42836.1    8\n",
      "BAF57210.1    7\n",
      "Q0VLQ1        6\n",
      "             ..\n",
      "P41365        1\n",
      "AAA33335.1    1\n",
      "A0A075B5G4    1\n",
      "2FX5_A        1\n",
      "ADV92525.1    1\n",
      "Name: count, Length: 162, dtype: int64\n",
      "No. of distinct values = 162\n",
      "No. of missing values = 1381\n",
      "\n",
      "Sequence\n",
      "MRIRRQAGTGARASMARAIGVMTTALAVLVGAVGGVAGAEVSTAQDNPYERGPDPTEDSIEAIRGPFSVATERVSSFASGFGGGTIYYPRETDEGTFGAVAVAPGFTASQGSMSWYGERVASQGFIVFTIDTNTRLDQPGQRGRQLLAALDYLVERSDRKVRERLDPNRLAVMGHSMGGGGSLEATVMRPSLKASIPLTPWNLDKTWGQVQVPTFIIGAELDTIASVRTHAKPFYESLPSSLPKAYMELDGATHFAPNIPNTTIAKYVISWLKRFVDEDTRYSQFLCPNPTDRAIEEYRSTCPY                                                                                                                                                                                                  16\n",
      "MFIMIKKSELAKAIIVTGALVFSIPTLAEVTLSETTVSSIKSEATVSSTKKALPATPSDCIADSKITAVALSDTRDNGPFSIRTKRISRQSAKGFGGGTIHYPTNASGCGLLGAIAVVPGYVSYENSIKWWGPRLASWGFVVITINTNSIYDDPDSRAAQLNAALDNMIADDTVGSMIDPKRLGAIGWSMGGGGALKLATERSTVRAIMPLAPYHDKSYGEVKTPTLVIACEDDRIAETKKYANAFYKNAIGPKMKVEVNNGSHFCPSYRFNEILLSKPGIAWMQRYINNDTRFDKFLCANENYSKSPRISAYDYKDCP                                                                                                                                                                                    9\n",
      "MAFNFIRAAAAGAAMALCGVGSVHAAVNLPALKIDKTQTTVSGLSSGGFMAVQLHVAYSATFAKGAGVVAGGPFYCAEGSIVNATGRCMASPAGIPTSTLVSTTNTWASQGVIDPVANLQNSKVYLFSGTLDSVVKTGVMDALRTYYNSFVPAANVVYKKDIAAEHAMVTDDYGNACSTKGAPYISDCNFDLAGAMLQHLYGTLNARNNATLPTGNYIEFNQSEFITNHGMATTGWAYVPQACQAGGTATCKLHVVLHGCKQNIGDVQQQYVRNTGYNRWADTNNIVMLYPQTSTAATNSCWDWWGYDSANYSKKSGPQMAAIKAMVDRVSSGTGGTTPPDPVALPAPTGVSTSGATASSMAIGWAAVMGAASYNVYRNANKVNALPVTATSYTDTGLAASTTYSWTVRAADANGAEGATSAAASGTTLAASGGGTATCTTASNYAHTLAGRAYAAGGYTYALGSNQNMGLWNVFVTNTLKQTSTNYYVIGTCP     8\n",
      "MRGVWRYMPVYYYKCMLIPLLTFTKISIISKPQESRISLIICRTMPTVATFLGLAISGAAMASGAALGAYNVDPNSISVSGLSSGGFMSAQLGVAYSDTFKVGFGVFAGGPYDCARGQSYTTCMYNQNPSITTPVANMKSWSGNKINPVSNLQSRKIYMWTGTADTTVGPNVMSQLKTQLANFASAANVSYITTSGAAHTFPTDFDAAGDNSCSSSVSPYISNCQCDGAGAVLQWMYGPLNARNPGTLSGSIVSFSQTGEYGASGMDTTGYLYVPRACQPGSSTVCKLHVALHGCKQSYSMIGSKFVSNTGYNMWADTNDIIILYPQAVADNTMHTIWTGMPLPNPNGCWDWVGWYGANADQVGGVQMAAIVNQVARVVSGYGAGSSSSTTAATPTTTTGSITTTTTTTATVTTTAVAPLYGQCGGIGWTGPTACATGVCTAYSPYYAQCLLIV                                             7\n",
      "MLLKKRWLIFISFLLAFTLLIPTAASASEKHYKPNIALEPIENSEGNEHPIILVHGLGGFGRDELGGIIKMWGGIHDIEKKLREKGYKVYTAAVGPVSSNRDRAIELYYQIKGGTVDYGEAHAKKYGHDRYGRTYPGFYPEWGEINPKTGKPNKVHLIGHSMGGQTIRTLAQLLYEGDPEEHKTGGNDISPLLSGEKQPWLHSVLSISSPHDGSTATYLVNDVIPIIQELVIGAAIFAGNIDQNLYDFKLDHWGIKKRPGESFHSYVQRVRNSPGWKTKDTANWDLKPEGAYELNRWVKAQPDVYYFSVSNTQSRRSLLTGYYVPDLFMNPFLHPTAYYIGSKTFRKSNFVLDKTWWENDGLVSVKAMKGPNIGSNDVIVEYNGTPRKGVWNHLGTMRQFDHLDIIGWGVRDVTSWYEDVARFLYSLPDDY                                                                    7\n",
      "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  ..\n",
      "MTTRPESLGLSAERLKRLDAHFDAKYVQTGKLPFAMTLVSRKGEVAHHGFSGLMDEARGKKVEADSLFRIYSMTKPLTSVLFMMLVEEGKVALDDPVHSLIPSWKNLGVYAGGWDMMASKPCARPMQMVDLLRHTSGLTYGFQQRTNVDAAYRKHNIGVFERQISLDATIEALAALPLEFSPGEAWNYSVSTDILGYLIGKIEGRPFEEVLKARLTGPLGMDDTDFQVKASERGRFAACSAINAKGERVLNDDPEKSSYLQPPVLVSGGGGLVSTAGDYLKFCQMMLNGGEANGTRFLSPKTIALMTSNHLPGGVDLPQLSKSLFSEVAYDGVGFGLGFSVTMDPAKTLLPGTAGEYAWGGAASTAFWIDPKEELIVIFMTQLIPSSFYPIRRELRTLAYSAITESFV                                                                                           1\n",
      "MRTVDGRIGMVERRTRLVSLLAAAALAVSLSGCSATTVLNTLEPRWDVAATRDVAYAPGPRHDVDVYAPKKIGPHTPVVVFIYGGGWDSGEKSKYAFVGSALASHGYLVFIPNYRIYPEAHYPDFLQDCALAVRWAKDHAAQYGGDPGELFLMGHSAGAYNAAMLATDPQWLGAVGMDPHRDLKGMVGLAGPYDFLPLQSDELKAIFGAVGQEPASQPINHVDGREPPMFLAHDLGDTVVYPKNTINMAAKIQAAGGEVETKYYKGLSHALMIGVFAAPLRFLGPVFRDSTQFIDAHASSSASKP                                                                                                                                                                                                  1\n",
      "MNEAMIRGVLEEAVASGGIAGAVVTVTDANATLFQLAVGRQSVAAPDPMRADSVFWIASMTKAITTVAALQLVEQGKLTLDEPIGDVLPELAEPVVLKGFGADGAPVVEPAQAKLTLKHLLTHTSGFSYEFANGDLARYLERTGHPSAATGLKAGLRQPLMFEPGARWEYGIGIDWAGQAVEAASGMTLDAYFAAYVTGPLGMTDTLFLPRADQAARRVAMHQRQPDGSLTAVPFEPRPAPEYLAGGAGLYSTAPDYAAFLRMILNKGMGLLKPETVAAMGTNQIGALRAGKIGSVNPGLTAEADFYPGMDSKWGLGFLINPEKGSFGRSADSLTWAGLPNCYYWIDREKGIGAVILMQSLPSGDMGALKAAAGFEYALYASL                                                                                                                    1\n",
      "MKKVIVFLVFMLFCTATYAQKVSKVVFFGDSLTDNGNLFRLVELIPKTPPYYQGRFSNGPTWAENLGEYFVKTNGAKYEIYAYAGATAAAHNIFETLAPTLLVDQVARYLETNPSDDKSKVLFVFWIGANDYLFLPTGNADELTTAAIQKTANGITTLLDQGAQNFLILDLPDLSKTPFAADLNMVSQMGSFTILHNQKLASLVDKMQKEHPQAQLMLFQLLQFSELIENPAKYNAKYHLNLSNTRAACWQGSVFDDIDTDNLANKTNKAISNPQPEVKEPILCEHPEQYVFFDHLHPTATVHQLLAALVEEAMNNRWS                                                                                                                                                                                    1\n",
      "MANPYERGPNPTDALLEARSGPFSVSEENVSRFGADGFGGGTIYYPRENNTYGAVAISPGYTGTQASVAWLGERIASHGFVVITIDTNTTLDQPDSRARQLNAALDYMINDASSAVRSRIDSSRLAVMGHSMGGGGTLRLASQRPDLKAAIPLTPWHLNKNWSSVRVPTLIIGADLDTIAPVLTHARPFYNSLPTSISKAYLELDGATHFAPNIPNKIIGKYSVAWLKRFVDNDTRYTQFLCPGPRDGLFGEVEEYRSTCPFALELEHHHHHH                                                                                                                                                                                                                                  1\n",
      "Name: count, Length: 199, dtype: int64\n",
      "No. of distinct values = 199\n",
      "No. of missing values = 1317\n",
      "\n",
      "Year\n",
      "2021.0    275\n",
      "2018.0    171\n",
      "2017.0    127\n",
      "1999.0     81\n",
      "2012.0     79\n",
      "2022.0     78\n",
      "2016.0     71\n",
      "2014.0     54\n",
      "2008.0     52\n",
      "2011.0     50\n",
      "1998.0     48\n",
      "2015.0     45\n",
      "2000.0     45\n",
      "2019.0     45\n",
      "2013.0     43\n",
      "2010.0     41\n",
      "2009.0     41\n",
      "2004.0     40\n",
      "2020.0     35\n",
      "2003.0     30\n",
      "2007.0     29\n",
      "2005.0     27\n",
      "2002.0     25\n",
      "2006.0     23\n",
      "1995.0     21\n",
      "1997.0     18\n",
      "2001.0     17\n",
      "1996.0     16\n",
      "1992.0     15\n",
      "1991.0     12\n",
      "1987.0     11\n",
      "1994.0      9\n",
      "1983.0      5\n",
      "1993.0      5\n",
      "1988.0      5\n",
      "1990.0      3\n",
      "1975.0      3\n",
      "1986.0      2\n",
      "1974.0      1\n",
      "1977.0      1\n",
      "1989.0      1\n",
      "Name: count, dtype: int64\n",
      "No. of distinct values = 41\n",
      "No. of missing values = 1\n",
      "\n",
      "Evidence\n",
      "Clear zone                                474\n",
      "Spectrophotometry                          55\n",
      "CO2;Clear zone                             55\n",
      "HPLC                                       55\n",
      "Weight loss                                49\n",
      "                                         ... \n",
      "MS;Clear zone;SEM;Weight loss;              1\n",
      "Clear zone;TOC;SEM                          1\n",
      "GPC;FTIR                                    1\n",
      "SEM;GPC;FTIR;Spectrophotometry;CO2;NMR      1\n",
      "FTIR-ATR;NMR;SEM                            1\n",
      "Name: count, Length: 296, dtype: int64\n",
      "No. of distinct values = 296\n",
      "No. of missing values = 45\n",
      "\n",
      "Plastic used\n",
      "Impranil                                                                                                                                                                                                                                                                                                                                                                                            247\n",
      "PHB granules                                                                                                                                                                                                                                                                                                                                                                                         28\n",
      "Amorphous PET film (Goodfellow Product ES30-FM-000145) and crystalline PET powder (Goodfellow Product ES30-PD-006031) were purchased from Goodfellow Corporation (USA).                                                                                                                                                                                                                              24\n",
      "PLA granules                                                                                                                                                                                                                                                                                                                                                                                         24\n",
      "Poly (3-hydroxybutric acid) (P(3HB) (Mn = 6.8 × 10^4, Mn/Mw = 2.6)                                                                                                                                                                                                                                                                                                                                   16\n",
      "                                                                                                                                                                                                                                                                                                                                                                                                   ... \n",
      "Aqueous PUR dispersion                                                                                                                                                                                                                                                                                                                                                                                1\n",
      "PCL was prepared by ring opening polymerisation of E-caprolactone in benzene in a nitrogen atmosphere at 60 °C with a diethylzinc-water catalyst system.                                                                                                                                                                                                                                              1\n",
      "High density PE                                                                                                                                                                                                                                                                                                                                                                                       1\n",
      "Microbial cultures were cultivated in liquid nutrient medium devoid of carbon. The source of this element for the microor ganisms was sectioned HDPE polyethylene foil, the polymeric material comprising in its chemical structure only carbon and hydrogen in the form of unbranched hydrocarbon chain. The biological material was cultivated on solidified Luria- Bertani Broth (LB) medium.      1\n",
      "Ecoflex is a commercial copolymer (BASF, Land Rheinland-Pfalz, Germany), which consists of 1,4-butanediol, adipic acid, and terephthalic acid                                                                                                                                                                                                                                                         1\n",
      "Name: count, Length: 553, dtype: int64\n",
      "No. of distinct values = 553\n",
      "No. of missing values = 127\n",
      "\n",
      "Manufacturer\n",
      " Covestro (Leverkusen, Germany)                    220\n",
      "Sigma Aldrich                                      193\n",
      "Showa Denko                                         79\n",
      "Mitsubishi Gas Chemical                             43\n",
      "Bayer                                               42\n",
      "                                                  ... \n",
      "Colora                                               1\n",
      "Colleague Hareware Co., Ltd. (Shanghai, China).      1\n",
      "Excelcier                                            1\n",
      "Crystal©                                             1\n",
      "Carry and Packers Plastics                           1\n",
      "Name: count, Length: 134, dtype: int64\n",
      "No. of distinct values = 134\n",
      "No. of missing values = 469\n",
      "\n",
      "Analitical grade\n",
      "Yes    488\n",
      "No     457\n",
      "NA       4\n",
      "Name: count, dtype: int64\n",
      "No. of distinct values = 3\n",
      "No. of missing values = 752\n",
      "\n",
      "Thermophilic conditions\n",
      "No     1475\n",
      "Yes     172\n",
      "Name: count, dtype: int64\n",
      "No. of distinct values = 2\n",
      "No. of missing values = 54\n",
      "\n",
      "Isolation sample type\n",
      "Soil                                       441\n",
      "Culture collection                         388\n",
      "Plastic debris                             108\n",
      "Water                                       42\n",
      "Compost                                     29\n",
      "Sediment                                    26\n",
      "Sludge                                      25\n",
      "Film                                        12\n",
      "PU coupons                                  11\n",
      "Soybean rhizosphere                          8\n",
      "Cow dung                                     8\n",
      "Marine                                       8\n",
      "Freshwater                                   8\n",
      "PVC film                                     6\n",
      "Leaf                                         6\n",
      "Barley                                       6\n",
      "Sediment                                     5\n",
      "Fiber                                        4\n",
      "Seawater                                     4\n",
      "Cacao pods                                   3\n",
      "Aircraft                                     3\n",
      "Plastic film                                 3\n",
      "Dye effluent enriched soil                   3\n",
      "Seeds                                        3\n",
      "Waxworm's gut                                3\n",
      "Root nodules                                 2\n",
      "Colllection                                  2\n",
      "Sand                                         2\n",
      "Basidiocarp of mushrooms                     2\n",
      "Algae                                        2\n",
      "Activated sludge                             2\n",
      "Wastewater                                   2\n",
      "Bag                                          2\n",
      "PVF foams                                    2\n",
      "Washed rind cheese                           2\n",
      "Digester sludge                              2\n",
      "Deep sea                                     2\n",
      "LDPE film                                    2\n",
      "Gut                                          2\n",
      "Mealworm's gut                               1\n",
      "Catheter                                     1\n",
      "Water;Soil                                   1\n",
      "Ice                                          1\n",
      "Petroleum-soil                               1\n",
      "Woody plants                                 1\n",
      "Garbage                                      1\n",
      "Sponge                                       1\n",
      "Babassu palm                                 1\n",
      "Waste                                        1\n",
      "Soil                                         1\n",
      "Foci of corrosion of fiberglass СТР-4ТР      1\n",
      "Sewater                                      1\n",
      "Spacesuit                                    1\n",
      "Nepenthes ampullaria                         1\n",
      "Barrel with fruit test                       1\n",
      "Wall                                         1\n",
      "Wood                                         1\n",
      "Sewage sludge                                1\n",
      "PBS Film                                     1\n",
      "Name: count, dtype: int64\n",
      "No. of distinct values = 59\n",
      "No. of missing values = 490\n",
      "\n",
      "Isolation environment\n",
      "Culture collection            434\n",
      "Soil                          374\n",
      "Plastic waste dumping site    135\n",
      "Compost                        70\n",
      "Marine                         67\n",
      "Sewage/Sludge                  45\n",
      "Landfill                       36\n",
      "River/Lake                     29\n",
      "Plant associated               27\n",
      "Mangrove                       21\n",
      "Others                         11\n",
      "Hot spring                      8\n",
      "Animal associated               7\n",
      "Deep sea                        1\n",
      "Name: count, dtype: int64\n",
      "No. of distinct values = 14\n",
      "No. of missing values = 436\n",
      "\n",
      "Isolation location\n",
      "Japan            194\n",
      "India            174\n",
      "France           116\n",
      "Slovakia          52\n",
      "Thailand          45\n",
      "                ... \n",
      "Israel             1\n",
      "Thaiti             1\n",
      "Ukraine            1\n",
      "Bay of Bengal      1\n",
      "EUA                1\n",
      "Name: count, Length: 66, dtype: int64\n",
      "No. of distinct values = 66\n",
      "No. of missing values = 653\n",
      "\n",
      "Degradation extrapolated from enzyme\n",
      "No     1425\n",
      "Yes     271\n",
      "Name: count, dtype: int64\n",
      "No. of distinct values = 2\n",
      "No. of missing values = 5\n",
      "\n",
      "Paper id\n",
      "PETcan102    1\n",
      "PETcan202    1\n",
      "PETcan712    1\n",
      "PETcan711    1\n",
      "PETcan708    1\n",
      "PETcan702    1\n",
      "PETcan701    1\n",
      "PETcan611    1\n",
      "PETcan606    1\n",
      "PETcan504    1\n",
      "PETcan503    1\n",
      "PETcan412    1\n",
      "PETcan409    1\n",
      "PETcan407    1\n",
      "PETcan405    1\n",
      "PETcan403    1\n",
      "PETcan401    1\n",
      "PETcan307    1\n",
      "PETcan305    1\n",
      "PETcan301    1\n",
      "PETcan214    1\n",
      "PETcan211    1\n",
      "PETcan204    1\n",
      "PETcan716    1\n",
      "Name: count, dtype: int64\n",
      "No. of distinct values = 24\n",
      "No. of missing values = 1677\n",
      "\n",
      "lineage\n",
      "superkingdom:Bacteria,species:uncultured bacterium,                                                                                                                           57\n",
      "superkingdom:Bacteria,phylum:Actinobacteria,class:Actinomycetia,order:Streptomycetales,family:Streptomycetaceae,genus:Streptomyces,species:Streptomyces sp.,                  43\n",
      "superkingdom:Eukaryota,phylum:Ascomycota,class:Eurotiomycetes,order:Eurotiales,family:Aspergillaceae,genus:Aspergillus,species:Aspergillus niger,                             40\n",
      "superkingdom:Bacteria,phylum:Proteobacteria,class:Gammaproteobacteria,order:Pseudomonadales,family:Pseudomonadaceae,genus:Pseudomonas,species:Pseudomonas sp.,                38\n",
      "superkingdom:Bacteria,phylum:Firmicutes,class:Bacilli,order:Bacillales,family:Bacillaceae,genus:Bacillus,species:Bacillus sp. (in: Bacteria),                                 28\n",
      "                                                                                                                                                                              ..\n",
      "superkingdom:Eukaryota,phylum:Ascomycota,class:Eurotiomycetes,order:Eurotiales,family:Aspergillaceae,genus:Penicillium,species:Penicillium griseofulvum,                       1\n",
      "superkingdom:Eukaryota,phylum:Ascomycota,class:Eurotiomycetes,order:Eurotiales,family:Aspergillaceae,genus:Penicillium,species:Penicillium glabrum,                            1\n",
      "superkingdom:Eukaryota,phylum:Ascomycota,class:Eurotiomycetes,order:Eurotiales,family:Aspergillaceae,genus:Penicillium,species:Penicillium daleae,                             1\n",
      "superkingdom:Eukaryota,phylum:Ascomycota,class:Eurotiomycetes,order:Eurotiales,family:Aspergillaceae,genus:Penicillium,species:Penicillium chermesinum,                        1\n",
      "superkingdom:Bacteria,phylum:Actinobacteria,class:Actinomycetia,order:Pseudonocardiales,family:Pseudonocardiaceae,genus:Saccharopolyspora,species:Saccharopolyspora flava,     1\n",
      "Name: count, Length: 586, dtype: int64\n",
      "No. of distinct values = 586\n",
      "No. of missing values = 0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Microorganism</th>\n",
       "      <th>Tax ID</th>\n",
       "      <th>Plastic</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Enzyme</th>\n",
       "      <th>Enzyme ID</th>\n",
       "      <th>Database Enzyne Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>GenbankID</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>...</th>\n",
       "      <th>Plastic used</th>\n",
       "      <th>Manufacturer</th>\n",
       "      <th>Analitical grade</th>\n",
       "      <th>Thermophilic conditions</th>\n",
       "      <th>Isolation sample type</th>\n",
       "      <th>Isolation environment</th>\n",
       "      <th>Isolation location</th>\n",
       "      <th>Degradation extrapolated from enzyme</th>\n",
       "      <th>Paper id</th>\n",
       "      <th>lineage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ralstonia pickettii</td>\n",
       "      <td>329</td>\n",
       "      <td>PHB</td>\n",
       "      <td>Yukawa, H., Uchida, Y., Kohama, K., &amp; Kurusu, ...</td>\n",
       "      <td>PHB depolymerase</td>\n",
       "      <td>1.0</td>\n",
       "      <td>00001 | PHB depolymerase | Ralstonia pickettii...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>BAA04986.1</td>\n",
       "      <td>MKHPYGYRWHWLYALVVTLMTALATFSAHAAVTAGPGAWSSQQTWA...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Soil</td>\n",
       "      <td>Soil</td>\n",
       "      <td>Japan</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>superkingdom:Bacteria,phylum:Proteobacteria,cl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ralstonia pickettii</td>\n",
       "      <td>329</td>\n",
       "      <td>PHA</td>\n",
       "      <td>Yukawa, H., Uchida, Y., Kohama, K., &amp; Kurusu, ...</td>\n",
       "      <td>PHB depolymerase</td>\n",
       "      <td>1.0</td>\n",
       "      <td>00001 | PHB depolymerase | Ralstonia pickettii...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>BAA04986.1</td>\n",
       "      <td>MKHPYGYRWHWLYALVVTLMTALATFSAHAAVTAGPGAWSSQQTWA...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Soil</td>\n",
       "      <td>Soil</td>\n",
       "      <td>Japan</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>superkingdom:Bacteria,phylum:Proteobacteria,cl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Pseudomonas fluorescens</td>\n",
       "      <td>294</td>\n",
       "      <td>PHO</td>\n",
       "      <td>Schirmer, A., &amp; Jendrossek, D. (1994). Molecul...</td>\n",
       "      <td>PHA depolymerase</td>\n",
       "      <td>2.0</td>\n",
       "      <td>00002 | PHA depolymerase | Pseudomonas fluores...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>AAA64538.1</td>\n",
       "      <td>MPLRTLLCGLLLAVCLGQHALAASRCSERPRTLLRPAEVSCSYQST...</td>\n",
       "      <td>...</td>\n",
       "      <td>P(3HO) was isolated from P. oleovorans grown ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>superkingdom:Bacteria,phylum:Proteobacteria,cl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Pseudomonas fluorescens</td>\n",
       "      <td>294</td>\n",
       "      <td>PHA</td>\n",
       "      <td>Schirmer, A., &amp; Jendrossek, D. (1994). Molecul...</td>\n",
       "      <td>PHA depolymerase</td>\n",
       "      <td>2.0</td>\n",
       "      <td>00002 | PHA depolymerase | Pseudomonas fluores...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>AAA64538.1</td>\n",
       "      <td>MPLRTLLCGLLLAVCLGQHALAASRCSERPRTLLRPAEVSCSYQST...</td>\n",
       "      <td>...</td>\n",
       "      <td>P(3HO) was isolated from P. oleovorans grown ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>superkingdom:Bacteria,phylum:Proteobacteria,cl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Paucimonas lemoignei</td>\n",
       "      <td>29443</td>\n",
       "      <td>PHB</td>\n",
       "      <td>Jendrossek, D. I. E. T. E. R., Frisse, A. N. D...</td>\n",
       "      <td>PHB depolymerase A</td>\n",
       "      <td>3.0</td>\n",
       "      <td>00003 | PHB depolymerase-A | Paucimonas lemoig...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>AAA65705.1</td>\n",
       "      <td>MRNTLKAAFKLGVISAALLAPFATQAATAGPGAWSSQQTWAADSVN...</td>\n",
       "      <td>...</td>\n",
       "      <td>The homopolyester PHB was isolated from sodium...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>Culture collection</td>\n",
       "      <td>Culture collection</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>superkingdom:Bacteria,phylum:Proteobacteria,cl...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Microorganism  Tax ID Plastic  \\\n",
       "0      Ralstonia pickettii     329     PHB   \n",
       "1      Ralstonia pickettii     329     PHA   \n",
       "2  Pseudomonas fluorescens     294     PHO   \n",
       "3  Pseudomonas fluorescens     294     PHA   \n",
       "4     Paucimonas lemoignei   29443     PHB   \n",
       "\n",
       "                                                 Ref              Enzyme  \\\n",
       "0  Yukawa, H., Uchida, Y., Kohama, K., & Kurusu, ...    PHB depolymerase   \n",
       "1  Yukawa, H., Uchida, Y., Kohama, K., & Kurusu, ...    PHB depolymerase   \n",
       "2  Schirmer, A., & Jendrossek, D. (1994). Molecul...    PHA depolymerase   \n",
       "3  Schirmer, A., & Jendrossek, D. (1994). Molecul...    PHA depolymerase   \n",
       "4  Jendrossek, D. I. E. T. E. R., Frisse, A. N. D...  PHB depolymerase A   \n",
       "\n",
       "   Enzyme ID                               Database Enzyne Name Gene  \\\n",
       "0        1.0  00001 | PHB depolymerase | Ralstonia pickettii...  Yes   \n",
       "1        1.0  00001 | PHB depolymerase | Ralstonia pickettii...  Yes   \n",
       "2        2.0  00002 | PHA depolymerase | Pseudomonas fluores...  Yes   \n",
       "3        2.0  00002 | PHA depolymerase | Pseudomonas fluores...  Yes   \n",
       "4        3.0  00003 | PHB depolymerase-A | Paucimonas lemoig...  Yes   \n",
       "\n",
       "    GenbankID                                           Sequence  ...  \\\n",
       "0  BAA04986.1  MKHPYGYRWHWLYALVVTLMTALATFSAHAAVTAGPGAWSSQQTWA...  ...   \n",
       "1  BAA04986.1  MKHPYGYRWHWLYALVVTLMTALATFSAHAAVTAGPGAWSSQQTWA...  ...   \n",
       "2  AAA64538.1  MPLRTLLCGLLLAVCLGQHALAASRCSERPRTLLRPAEVSCSYQST...  ...   \n",
       "3  AAA64538.1  MPLRTLLCGLLLAVCLGQHALAASRCSERPRTLLRPAEVSCSYQST...  ...   \n",
       "4  AAA65705.1  MRNTLKAAFKLGVISAALLAPFATQAATAGPGAWSSQQTWAADSVN...  ...   \n",
       "\n",
       "                                        Plastic used Manufacturer  \\\n",
       "0                                                NaN          NaN   \n",
       "1                                                NaN          NaN   \n",
       "2   P(3HO) was isolated from P. oleovorans grown ...          NaN   \n",
       "3   P(3HO) was isolated from P. oleovorans grown ...          NaN   \n",
       "4  The homopolyester PHB was isolated from sodium...          NaN   \n",
       "\n",
       "  Analitical grade Thermophilic conditions Isolation sample type  \\\n",
       "0              NaN                     NaN                  Soil   \n",
       "1              NaN                     NaN                  Soil   \n",
       "2              Yes                      No                   NaN   \n",
       "3              Yes                      No                   NaN   \n",
       "4              Yes                      No    Culture collection   \n",
       "\n",
       "  Isolation environment Isolation location  \\\n",
       "0                  Soil              Japan   \n",
       "1                  Soil              Japan   \n",
       "2                   NaN                NaN   \n",
       "3                   NaN                NaN   \n",
       "4    Culture collection                NaN   \n",
       "\n",
       "  Degradation extrapolated from enzyme Paper id  \\\n",
       "0                                  Yes      NaN   \n",
       "1                                  Yes      NaN   \n",
       "2                                  Yes      NaN   \n",
       "3                                  Yes      NaN   \n",
       "4                                   No      NaN   \n",
       "\n",
       "                                             lineage  \n",
       "0  superkingdom:Bacteria,phylum:Proteobacteria,cl...  \n",
       "1  superkingdom:Bacteria,phylum:Proteobacteria,cl...  \n",
       "2  superkingdom:Bacteria,phylum:Proteobacteria,cl...  \n",
       "3  superkingdom:Bacteria,phylum:Proteobacteria,cl...  \n",
       "4  superkingdom:Bacteria,phylum:Proteobacteria,cl...  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_microbes = pd.read_csv('../Data/__plasticdb_microbes.tsv', delimiter='\\t')\n",
    "desc_df(df_microbes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Enzyme ID</th>\n",
       "      <th>Degradation extrapolated from enzyme</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2.0</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2.0</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3.0</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1696</th>\n",
       "      <td>204.0</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1697</th>\n",
       "      <td>205.0</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1698</th>\n",
       "      <td>206.0</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1699</th>\n",
       "      <td>207.0</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1700</th>\n",
       "      <td>208.0</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>379 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Enzyme ID Degradation extrapolated from enzyme\n",
       "0           1.0                                  Yes\n",
       "1           1.0                                  Yes\n",
       "2           2.0                                  Yes\n",
       "3           2.0                                  Yes\n",
       "4           3.0                                   No\n",
       "...         ...                                  ...\n",
       "1696      204.0                                  Yes\n",
       "1697      205.0                                  Yes\n",
       "1698      206.0                                  Yes\n",
       "1699      207.0                                  Yes\n",
       "1700      208.0                                  Yes\n",
       "\n",
       "[379 rows x 2 columns]"
      ]
     },
     "execution_count": 259,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "useful_features = [\"Enzyme ID\", 'Degradation extrapolated from enzyme']\n",
    "df_microbes_useful = df_microbes[useful_features]\n",
    "df_microbes_useful = df_microbes_useful[~df_microbes_useful['Enzyme ID'].isna()]\n",
    "df_microbes_useful = df_microbes_useful[~df_microbes_useful['Degradation extrapolated from enzyme'].isna()]\n",
    "df_microbes_useful"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>degradation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>203 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     degradation\n",
       "id              \n",
       "1              1\n",
       "2              1\n",
       "3              0\n",
       "4              0\n",
       "5              0\n",
       "..           ...\n",
       "204            1\n",
       "205            1\n",
       "206            1\n",
       "207            1\n",
       "208            1\n",
       "\n",
       "[203 rows x 1 columns]"
      ]
     },
     "execution_count": 260,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enzyme_degradation = {}\n",
    "for index, row in df_microbes_useful.iterrows():\n",
    "    enzyme_id = row['Enzyme ID']\n",
    "    degradation = int(row['Degradation extrapolated from enzyme'] == 'Yes')\n",
    "    if (\n",
    "        enzyme_id not in enzyme_degradation \n",
    "        or enzyme_id in enzyme_degradation \n",
    "        and enzyme_degradation[int(enzyme_id)] == 0\n",
    "    ): enzyme_degradation[int(enzyme_id)] = degradation\n",
    "enzyme_degradation = {\n",
    "    \"id\": list(enzyme_degradation.keys()), \n",
    "    \"degradation\": list(enzyme_degradation.values())\n",
    "}\n",
    "df_enzyme_degradation = pd.DataFrame(enzyme_degradation)\n",
    "df_enzyme_degradation.set_index('id', inplace=True, drop=True)\n",
    "df_enzyme_degradation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>source</th>\n",
       "      <th>plastic</th>\n",
       "      <th>n_plastics</th>\n",
       "      <th>aa_seq</th>\n",
       "      <th>degradation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>phb_depolymerase</td>\n",
       "      <td>ralstonia_pickettii</td>\n",
       "      <td>phb_pha</td>\n",
       "      <td>2</td>\n",
       "      <td>(M, K, H, P, Y, G, Y, R, W, H, W, L, Y, A, L, ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>pha_depolymerase</td>\n",
       "      <td>pseudomonas_fluorescens</td>\n",
       "      <td>pho_pha</td>\n",
       "      <td>2</td>\n",
       "      <td>(M, P, L, R, T, L, L, C, G, L, L, L, A, V, C, ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>phb_depolymerase_a</td>\n",
       "      <td>paucimonas_lemoignei</td>\n",
       "      <td>phb_pha</td>\n",
       "      <td>2</td>\n",
       "      <td>(M, R, N, T, L, K, A, A, F, K, L, G, V, I, S, ...</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>phb_depolymerase</td>\n",
       "      <td>paucimonas_lemoignei</td>\n",
       "      <td>phb_pha</td>\n",
       "      <td>2</td>\n",
       "      <td>(M, L, A, K, Q, I, K, K, A, N, S, R, S, T, L, ...</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>pha_depolymerase_b</td>\n",
       "      <td>paucimonas_lemoignei</td>\n",
       "      <td>phb_pha</td>\n",
       "      <td>2</td>\n",
       "      <td>(M, M, S, S, Q, T, T, Q, S, S, K, F, S, L, F, ...</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204</th>\n",
       "      <td>petase</td>\n",
       "      <td>thermobifida_fusca</td>\n",
       "      <td>pet</td>\n",
       "      <td>1</td>\n",
       "      <td>(M, A, A, N, P, Y, E, R, G, P, N, P, T, D, A, ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>petase</td>\n",
       "      <td>thermobifida_alba</td>\n",
       "      <td>pet</td>\n",
       "      <td>1</td>\n",
       "      <td>(M, A, N, P, Y, E, R, G, P, N, P, T, E, S, M, ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>petase</td>\n",
       "      <td>thermobifida_cellulosilytica</td>\n",
       "      <td>pet</td>\n",
       "      <td>1</td>\n",
       "      <td>(M, A, N, P, Y, E, R, G, P, D, P, T, Q, A, S, ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>petase</td>\n",
       "      <td>thermobifida_halotolerans</td>\n",
       "      <td>pet</td>\n",
       "      <td>1</td>\n",
       "      <td>(M, A, N, P, Y, E, R, G, P, N, P, T, N, S, S, ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>petase</td>\n",
       "      <td>thermobifida_cellulosilytica</td>\n",
       "      <td>pet</td>\n",
       "      <td>1</td>\n",
       "      <td>(M, A, N, P, Y, E, R, G, P, N, P, T, D, A, L, ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>208 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   name                        source  plastic  n_plastics  \\\n",
       "id                                                                           \n",
       "1      phb_depolymerase           ralstonia_pickettii  phb_pha           2   \n",
       "2      pha_depolymerase       pseudomonas_fluorescens  pho_pha           2   \n",
       "3    phb_depolymerase_a          paucimonas_lemoignei  phb_pha           2   \n",
       "4      phb_depolymerase          paucimonas_lemoignei  phb_pha           2   \n",
       "5    pha_depolymerase_b          paucimonas_lemoignei  phb_pha           2   \n",
       "..                  ...                           ...      ...         ...   \n",
       "204              petase            thermobifida_fusca      pet           1   \n",
       "205              petase             thermobifida_alba      pet           1   \n",
       "206              petase  thermobifida_cellulosilytica      pet           1   \n",
       "207              petase     thermobifida_halotolerans      pet           1   \n",
       "208              petase  thermobifida_cellulosilytica      pet           1   \n",
       "\n",
       "                                                aa_seq  degradation  \n",
       "id                                                                   \n",
       "1    (M, K, H, P, Y, G, Y, R, W, H, W, L, Y, A, L, ...          1.0  \n",
       "2    (M, P, L, R, T, L, L, C, G, L, L, L, A, V, C, ...          1.0  \n",
       "3    (M, R, N, T, L, K, A, A, F, K, L, G, V, I, S, ...          0.0  \n",
       "4    (M, L, A, K, Q, I, K, K, A, N, S, R, S, T, L, ...          0.0  \n",
       "5    (M, M, S, S, Q, T, T, Q, S, S, K, F, S, L, F, ...          0.0  \n",
       "..                                                 ...          ...  \n",
       "204  (M, A, A, N, P, Y, E, R, G, P, N, P, T, D, A, ...          1.0  \n",
       "205  (M, A, N, P, Y, E, R, G, P, N, P, T, E, S, M, ...          1.0  \n",
       "206  (M, A, N, P, Y, E, R, G, P, D, P, T, Q, A, S, ...          1.0  \n",
       "207  (M, A, N, P, Y, E, R, G, P, N, P, T, N, S, S, ...          1.0  \n",
       "208  (M, A, N, P, Y, E, R, G, P, N, P, T, D, A, L, ...          1.0  \n",
       "\n",
       "[208 rows x 6 columns]"
      ]
     },
     "execution_count": 261,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_proteins = pd.concat(\n",
    "    [df_proteins, df_enzyme_degradation],\n",
    "    axis=1,\n",
    "    join=\"outer\",\n",
    "    ignore_index=False,\n",
    "    keys=None,\n",
    "    levels=None,\n",
    "    names=None,\n",
    "    verify_integrity=False,\n",
    "    copy=True,\n",
    ")\n",
    "df_proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>source</th>\n",
       "      <th>plastic</th>\n",
       "      <th>n_plastics</th>\n",
       "      <th>aa_seq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>phb_depolymerase</td>\n",
       "      <td>ralstonia_pickettii</td>\n",
       "      <td>phb_pha</td>\n",
       "      <td>2</td>\n",
       "      <td>(M, K, H, P, Y, G, Y, R, W, H, W, L, Y, A, L, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>pha_depolymerase</td>\n",
       "      <td>pseudomonas_fluorescens</td>\n",
       "      <td>pho_pha</td>\n",
       "      <td>2</td>\n",
       "      <td>(M, P, L, R, T, L, L, C, G, L, L, L, A, V, C, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>pva_dehydrogenase</td>\n",
       "      <td>pseudomonas_sp.</td>\n",
       "      <td>pva</td>\n",
       "      <td>1</td>\n",
       "      <td>(M, Q, Q, N, I, E, R, N, Q, V, S, M, T, T, S, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>phb_depolymerase</td>\n",
       "      <td>comamonas_acidovorans</td>\n",
       "      <td>p3hp_p4hb_pea_pes_phb_pha</td>\n",
       "      <td>6</td>\n",
       "      <td>(M, A, F, N, F, I, R, A, A, A, A, G, A, A, M, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>phb_depolymerase</td>\n",
       "      <td>comamonas_testosteroni</td>\n",
       "      <td>phb_pha</td>\n",
       "      <td>2</td>\n",
       "      <td>(M, R, V, Q, S, W, R, S, G, V, A, A, L, A, L, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>petase</td>\n",
       "      <td>thermobifida_fusca</td>\n",
       "      <td>pet</td>\n",
       "      <td>1</td>\n",
       "      <td>(M, A, A, N, P, Y, E, R, G, P, N, P, T, D, A, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>petase</td>\n",
       "      <td>thermobifida_alba</td>\n",
       "      <td>pet</td>\n",
       "      <td>1</td>\n",
       "      <td>(M, A, N, P, Y, E, R, G, P, N, P, T, E, S, M, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>petase</td>\n",
       "      <td>thermobifida_cellulosilytica</td>\n",
       "      <td>pet</td>\n",
       "      <td>1</td>\n",
       "      <td>(M, A, N, P, Y, E, R, G, P, D, P, T, Q, A, S, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>petase</td>\n",
       "      <td>thermobifida_halotolerans</td>\n",
       "      <td>pet</td>\n",
       "      <td>1</td>\n",
       "      <td>(M, A, N, P, Y, E, R, G, P, N, P, T, N, S, S, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>petase</td>\n",
       "      <td>thermobifida_cellulosilytica</td>\n",
       "      <td>pet</td>\n",
       "      <td>1</td>\n",
       "      <td>(M, A, N, P, Y, E, R, G, P, N, P, T, D, A, L, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  name                        source  \\\n",
       "0     phb_depolymerase           ralstonia_pickettii   \n",
       "1     pha_depolymerase       pseudomonas_fluorescens   \n",
       "2    pva_dehydrogenase               pseudomonas_sp.   \n",
       "3     phb_depolymerase         comamonas_acidovorans   \n",
       "4     phb_depolymerase        comamonas_testosteroni   \n",
       "..                 ...                           ...   \n",
       "116             petase            thermobifida_fusca   \n",
       "117             petase             thermobifida_alba   \n",
       "118             petase  thermobifida_cellulosilytica   \n",
       "119             petase     thermobifida_halotolerans   \n",
       "120             petase  thermobifida_cellulosilytica   \n",
       "\n",
       "                       plastic  n_plastics  \\\n",
       "0                      phb_pha           2   \n",
       "1                      pho_pha           2   \n",
       "2                          pva           1   \n",
       "3    p3hp_p4hb_pea_pes_phb_pha           6   \n",
       "4                      phb_pha           2   \n",
       "..                         ...         ...   \n",
       "116                        pet           1   \n",
       "117                        pet           1   \n",
       "118                        pet           1   \n",
       "119                        pet           1   \n",
       "120                        pet           1   \n",
       "\n",
       "                                                aa_seq  \n",
       "0    (M, K, H, P, Y, G, Y, R, W, H, W, L, Y, A, L, ...  \n",
       "1    (M, P, L, R, T, L, L, C, G, L, L, L, A, V, C, ...  \n",
       "2    (M, Q, Q, N, I, E, R, N, Q, V, S, M, T, T, S, ...  \n",
       "3    (M, A, F, N, F, I, R, A, A, A, A, G, A, A, M, ...  \n",
       "4    (M, R, V, Q, S, W, R, S, G, V, A, A, L, A, L, ...  \n",
       "..                                                 ...  \n",
       "116  (M, A, A, N, P, Y, E, R, G, P, N, P, T, D, A, ...  \n",
       "117  (M, A, N, P, Y, E, R, G, P, N, P, T, E, S, M, ...  \n",
       "118  (M, A, N, P, Y, E, R, G, P, D, P, T, Q, A, S, ...  \n",
       "119  (M, A, N, P, Y, E, R, G, P, N, P, T, N, S, S, ...  \n",
       "120  (M, A, N, P, Y, E, R, G, P, N, P, T, D, A, L, ...  \n",
       "\n",
       "[121 rows x 5 columns]"
      ]
     },
     "execution_count": 262,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_proteins = df_proteins[df_proteins['degradation'] == 1].reset_index(drop=True)\n",
    "df_proteins.drop('degradation', axis=1, inplace=True)\n",
    "df_proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 263,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df_proteins.to_csv('../Data/__plasticdb_protein_sequences_clean.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "dissertation-tcd",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.3"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
